Library/Homebrew/cmd/update.rb

```diff
@@ -277,8 +277,7 @@ def pull!(options = {})
     reset_on_interrupt { safe_system "git", *args }
 
     @current_revision = read_current_revision
-
-    if @initial_branch != @upstream_branch && !@initial_branch.empty?
+    if ARGV.homebrew_developer? && @initial_branch != @upstream_branch && !@initial_branch.empty?
```

Comment:
Could we use a different flag or ENV variable for this? It is counter-intiative to have `HOMEBREW_DEVELOPER` uses a complete different update strategy.